clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/ServerStoreProxy.java

```diff
@@ -44,25 +51,45 @@ public String getCacheId() {
 
   @Override
   public Chain get(long key) {
-    // TODO: Implement ServerStoreProxyImpl.get
-    throw new UnsupportedOperationException("ServerStoreProxyImpl.get not implemented");
+    try {
+      EhcacheEntityResponse response = entity.invoke(EhcacheEntityMessage.getOperation(cacheId, key), false);
+      if (response != null && response.getType() == EhcacheEntityResponse.Type.GET_RESPONSE) {
+        return ((EhcacheEntityResponse.GetResponse)response).getChain();
+      }
```

Comment:
I meant for that also it should throw, because both cases should never happen, So if ti does not falls in if clause it should be an Exception